app/controllers/contacts.js

```diff
@@ -0,0 +1,144 @@
+import BaseController from "./base/base.js";
+import Contact from "./../models/contact.js";
+import User from "./../models/user.js";
+import SessionRepository from "./../repositories/session_repository.js";
+import { ACTIVE } from "./../store/session.js";
+import { ERROR_STATUES } from "./../validations/constants/errors.js";
+
+class ContactsController extends BaseController {
+  constructor() {
+    super();
+
+    this.sessionRepository = new SessionRepository(ACTIVE);
+  }
+
+  async contact_add(ws, data) {
+    const {
+      id: requestId,
+      contact_add: { email, phone },
+    } = data;
+    const currentUser = this.sessionRepository.getSessionUserId(ws);
+    const contactData = data.contact_add;
+
+    if (email) {
+      contactData.email = [];
+      const findedUsersList = await User.findAll(
+        { email: { $in: email.map((el) => el.email) } },
+        ["_id", "email"]
+      );
+
+      const findedUsersObj = {};
+      for (let i = 0; i < findedUsersList.length; i++) {
+        const obj = findedUsersList[i];
+        findedUsersObj[obj.email] = obj._id;
+      }
+
+      for (let i = 0; i < email.length; i++) {
+        const obj = email[i];
+        if (findedUsersObj[obj.value]) {
+          obj["matched_user_id"] = findedUsersObj[obj.value];
+        }
+
+        contactData.email.push(obj);
+      }
+    }
+
+    if (phone) {
```

Comment:
all this logic looks pretty the same to email

can we extract it into separate universal method?